src/main/java/com/powsybl/openloadflow/AbstractAcOuterLoopConfig.java

```diff
@@ -19,13 +21,27 @@
  */
 abstract class AbstractAcOuterLoopConfig implements AcOuterLoopConfig {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractAcOuterLoopConfig.class);
+
     protected AbstractAcOuterLoopConfig() {
     }
 
     protected static Optional<AcOuterLoop> createDistributedSlackOuterLoop(LoadFlowParameters parameters, OpenLoadFlowParameters parametersExt) {
         if (parameters.isDistributedSlack()) {
+            if (parametersExt.isAreaInterchangeControl()) {
+                LOGGER.info("DistributedSlack and AreaInterchangeControl outer loops are both enabled. DistributedSlack outerloop will not be created, slack will be distributed by the AreaInterchangeControl outerloop.");
```

Comment:
This will break if an explicit outerloop list is specified and if it contains DistributedSclackOuterLoop but not AreaInterchangeControlOuterloop.
This control should not be in this factory level.
Maybe be:
  this logic in DefaultAcOuterLoop (that creates outerloop based on parameters)
  a check in Explicit that you do not have both DistributedSclackOuterLoop and AreaInterchangeControlOuterloop at the same time because they are incompatible.

Reply:
i suggest another way to fix it in 6cc333a90c271a4448bce95b72d440602624d227

Reply:
I think your approach works as well.
One minor remark: in the new test you could explicitely set the param distribtuedSlack to true. This way you won't rely on the default value to test that the loop is actually filtered in the default configuration.